cedar-lean/Cedar/Validation/RequestEntityValidator.lean

```diff
@@ -0,0 +1,157 @@
+import Cedar.Validation.Validator
+import Cedar.Validation.Typechecker
+namespace Cedar.Validation
+
+open Cedar.Spec
+open Cedar.Data
+
+inductive RequestValidationError where
+| typeError (loc : String)
+
+abbrev RequestValidationResult := Except RequestValidationError Unit
+
+inductive EntityValidationError where
+| typeError (loc : String)
+
+abbrev EntityValidationResult := Except EntityValidationError Unit
+
+def instanceOfBoolType (b : Bool) (bty : BoolType) : Bool :=
+match b, bty with
+  | true, .tt => true
+  | false, .ff => true
+  | _, .anyBool => true
+  | _, _ => false
+
+def instanceOfEntityType (e : EntityUID) (ety : EntityType ) : Bool := ety == e.ty
+
+def instanceOfExtType (ext : Ext) (extty: ExtType) : Bool :=
+match ext, extty with
+  | .decimal _, .decimal => true
+  | .ipaddr _, .ipAddr => true
+  | _, _ => false
+
+def requiredAttributesPresent (r : Map Attr Value) (rty : Map Attr (Qualified CedarType)) (k : Attr) :=
```

Comment:
This is testing the presence of a single attribute.  Let's rename it to `requiredAttributePresent`.